### PR TITLE
Added Wait time to sending each Keys for Dynamic Macros function

### PR DESCRIPTION
**Pull request by FREEWING-JP**
## Description

When recording and playing back UI operations in the Dynamic Macros function, the UI may miss keystrokes because the keys are sent too fast. 
This problem can be solved by adding a wait time for each key transmission.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_dynamic_macros.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ There are a number of options added that should allow some additional degree of
 |`DYNAMIC_MACRO_SIZE`        |128             |Sets the amount of memory that Dynamic Macros can use. This is a limited resource, dependent on the controller.  |
 |`DYNAMIC_MACRO_USER_CALL`   |*Not defined*   |Defining this falls back to using the user `keymap.c` file to trigger the macro behavior.                        |
 |`DYNAMIC_MACRO_NO_NESTING`  |*Not Defined*   |Defining this disables the ability to call a macro from another macro (nested macros).                           | 
+|`DYNAMIC_MACRO_WAIT`        |*Not Defined*   |Sets the waiting time (ms unit) when sending each key.                                                           |
```
Comment (zvecr) [inline docs/feature_dynamic_macros.md:38]:
```suggestion
|`DYNAMIC_MACRO_DELAY`        |*Not Defined*   |Sets the waiting time (ms unit) when sending each key.                                                           |
```

File: quantum/process_keycode/process_dynamic_macro.c [hunk 2]
```diff
@@ -86,6 +86,9 @@ void dynamic_macro_play(keyrecord_t *macro_buffer, keyrecord_t *macro_end, int8_
     while (macro_buffer != macro_end) {
         process_record(macro_buffer);
         macro_buffer += direction;
+#ifdef DYNAMIC_MACRO_WAIT
+        wait_ms(DYNAMIC_MACRO_WAIT);
```
Comment (zvecr) [inline quantum/process_keycode/process_dynamic_macro.c:90]:
```suggestion
#ifdef DYNAMIC_MACRO_DELAY
        wait_ms(DYNAMIC_MACRO_DELAY);
```

File: docs/ja/feature_dynamic_macros.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ QMK はその場で作られた一時的なマクロをサポートします。
 | `DYNAMIC_MACRO_SIZE` | 128 | 動的マクロが使用できるメモリ量を設定します。これは限られたリソースであり、コントローラに依存します。 |
 | `DYNAMIC_MACRO_USER_CALL` | *定義なし* | これを定義すると、ユーザの `keymap.c` ファイルを使ってマクロが起動されます。 |
 | `DYNAMIC_MACRO_NO_NESTING` | *定義なし* | これを定義すると、別のマクロからマクロを呼び出す(入れ子になったマクロ)機能を無効にします。 |
+| `DYNAMIC_MACRO_WAIT` | *定義なし* | 各キーを送信する時の待ち時間（ms単位）を設定します。 |
```
Comment (zvecr) [inline docs/ja/feature_dynamic_macros.md:43]:
```suggestion
| `DYNAMIC_MACRO_DELAY` | *定義なし* | 各キーを送信する時の待ち時間（ms単位）を設定します。 |
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Most items in this area are `_DELAY` and not "wait".

**Review by zvecr — APPROVED** [review]

**Comment by drashna** [comment]
It may be worth having this set to use `TAP_CODE_DELAY` by default, or just use it altogether, rather than a separate define.

**Comment by FREEWING-JP** [comment]
I traced program .
dynamic_macro_play() -> process_record() -> process_record_handler() -> process_action() .
In process_action() function using TAP_CODE_DELAY .

Looks like It is not appropriate to put delay logic inside the dynamic_macro_play() function .
However, defining TAP_CODE_DELAY affects all process_record()/ process_actions() .

Hmm...

**Review by tzarc — APPROVED** [review]

**Comment by FREEWING-JP** [comment]
Thank you .

**Pull request merged**